### PR TITLE
prevent data race in pkg/plugins

### PR DESCRIPTION
**Pull request by vieux**
ping @anusha-ragunathan

### CODE REVIEW

**Comment by aaronlehmann** [comment]
LGTM

Thanks.

**Comment by aaronlehmann** [comment]
@vieux: Needs a rebase

Would love to get this merged, so I can update vendoring in swarmkit.

**Comment by mlaventure** [comment]
LGTM, still needs a rebase :)

**Comment by vieux** [comment]
I'll rebase asap

**Comment by vieux** [comment]
please @aaronlehmann @mlaventure PTAL

File: pkg/plugins/plugins.go [hunk 1]
```diff
@@ -41,11 +41,20 @@ type plugins struct {
...
+
 var (
-	storage          = plugins{plugins: make(map[string]*Plugin)}
-	extpointHandlers = make(map[string][]func(string, *Client))
+	storage  = plugins{plugins: make(map[string]*Plugin)}
+	handlers = extpointHandlers{}
 )
 
+func init() {
+	handlers.extpointHandlers = make(map[string][]func(string, *Client))
```
Comment (mlaventure) [inline pkg/plugins/plugins.go]: nit, any reason this isn't done when the variable is declared? (Just occurred to me)

**Review by mlaventure — CHANGES_REQUESTED** [review]

File: pkg/plugins/plugins.go (hunk 1 above)
Comment (vieux) [inline pkg/plugins/plugins.go]: no, thanks!

Comment (vieux) [inline pkg/plugins/plugins.go]: @mlaventure updated

**Review by mlaventure — APPROVED** [review]

**Comment by mlaventure** [comment]
LGTM

**Pull request merged**